woocommerce/woocommerce-android
# Product Creation with AI: Product Name subscreen functionality

Part of #9800 


(please ignore the branch name, it shouldn't have the `suggestion` name in it)
### Description


This adds these functionalities to the product name subscreen:
1. Product name editing,
2. Continue button

It also fixes a bug where previously entered product name is shown on the next (About product) screen's text field.

This **does not** include the functionality for "Suggest a name". I'll continue this task on https://github.com/woocommerce/woocommerce-android/pull/9808 instead.

### Testing instructions


1. Go to products list, tap "+" button,
2. Choose "Create a product with AI",
3. Make sure "Continue" button is disabled,
4. Enter some text into the text field, make sure that enables the "Continue" button,
5. Delete all text and make sure that disables the "Continue" button,
6. Enter some text again, tap "Continue", make sure it opens to the next screen ("About your product")
